### ISSUE TITLE
Picker: width calculation should be based on the longest item, not the selected item

### ISSUE THREAD

**Issue by ajhuntsman** [xamarin/Xamarin.Forms#1996]
### Description

In Xamarin Forms XAML, using `HorizontalOptions="Start"` or `HorizontalOptions="StartAndExpand"` for a `Picker` element may cause the final layout width of the Picker to be insufficient, and therefore truncate longer items upon selection.

### Steps to Reproduce

1. Follow the [recommended steps](https://blog.xamarin.com/new-bindable-picker-control-for-xamarin-forms/) from Xamarin for a basic Picker.
2. In the XAML, inside the `Picker` element, set its horizontal options to be `HorizontalOptions="Start"`.
3. In the view model, create a list of strings to display in the picker: `Red`, `Blue`, `Green`, `Magenta`, `Aquamarine`.
4. In the view model, set the initial selected index property to be `0` so that `Red` is selected by default.
5. Run the app, and note that the width of the picker exactly matches the width of the initial selected item, rather than matching the width of the longest item.
6. Tap on the picker and select `Magenta` or `Aquamarine`.

### Expected Behavior
The width of the Picker element at runtime should be wide enough to accommodate the longest value in the list ("best fit").

### Actual Behavior
The width of the Picker element at runtime is insufficient, which truncates longer values upon selection.

### Basic Information

- Version with issue: Xamarin.Forms (2.5.0.280555)
- Last known good version: (unknown)
- IDE: Visual Studio for Mac 7.4 Preview (7.4 build 1026)
- Platform Target Frameworks:
  - Android: 8.0
- Android Support Library Version: 26.1.0.1
- Nuget Packages:  
  - ReactiveUI.XamForms (8.0.0-alpha0117) 
  - Xam.Plugins.Settings (3.1.1)
  - Xam.Plugins.TextToSpeech (3.2.1)
  - Xam.Plugins.Vibrate (3.0.1)
  - Xamarin.Plugins.Clipboard (2.0.0)
- Affected Devices: Android devices

### Screenshots

![01_picker_width_matches_initial_value_red](https://user-images.githubusercontent.com/1542226/36806909-875887d0-1c8f-11e8-9a95-a3f197b3922f.png)

![02_picker_items](https://user-images.githubusercontent.com/1542226/36806926-9256d772-1c8f-11e8-913d-b663d95e6aa2.png)

![03_picker_width_truncates_longer_values](https://user-images.githubusercontent.com/1542226/36806932-9672d068-1c8f-11e8-966a-80df47fb501a.png)

**Comment by jassmith** [comment]
The native picker controls do not expose to us their maximum potential width unfortunately (when you measure the native controls the result you get is the size of the current selection).

A fix we can and should do is trigger a relayout when the picker control selection is changed.

**Comment by ajhuntsman** [comment]
Thanks for the fast response, @jassmith!

**Comment by ThomasOst** [comment]
Is also [Bug 57539](https://bugzilla.xamarin.com/show_bug.cgi?id=57539)

**Comment by charlesroddie** [comment]
This bug affects Android and iOS but not UWP.

**Comment by EJocys** [comment]
I am using this workaround:

```C#
private void Picker_SelectedIndexChanged(object sender, EventArgs e)
{
	ControlsHelper.InvalidateMeasure((Picker)sender);
}
```
It won't set size based on the longest item, but it will properly display all selected items.
InvalidateMeasure method inside my ControlsHelper.cs class (various Xamarin workarounds):

```C#
/// <summary>
/// Some controls fail to resize. Use this function as a workaround.
/// </summary>
/// <param name="control"></param>
public static void InvalidateMeasure(this VisualElement control)
{
	if (control == null)
		throw new ArgumentNullException(nameof(control));
	var methods = typeof(VisualElement).GetTypeInfo().DeclaredMethods;
	var method = methods.FirstOrDefault(x => x.Name == "InvalidateMeasure");
	if (method != null)
		method.Invoke(control, null);
}
```

**Comment by samhouts** [comment]
This issue doesn't seem to have had any activity in a long time. We're working on prioritizing issues and resolving them as quickly as we can. To help us get through the list, we would appreciate an update from you to let us know if this is still affecting you on the latest version of Xamarin.Forms, since it's possible that we may have resolved this as part of another related or duplicate issue. If we don't see any new activity on this issue in the next 30 days, we'll evaluate whether this issue should be closed. Thank you!

**Comment by Redth** [comment]
Since we haven't heard from you in more than 30 days, we hope this issue is no longer affecting you. If it is, please reopen this issue and provide the requested information so that we can look into it further. Thank you!

**Comment by kevcrooks** [comment]
I've just checked and we're having this issue on the latest version `5.0.0.2012`.

If we set a picker with short text and long text, then the picker won't resize on Android (but it will resize correctly on UWP). At the moment I think this means we need to set the picker to a fixed known length to avoid truncation.

I've attached details below, but given this has been closed for some time, shall I open a new issue?

Starting from a basic Xamarin.Forms app on all platforms:
```
        public App()
        {
            InitializeComponent();
            MainPage = new ContentPage
            {
                Content = new Picker
                {
                    ItemsSource = new List<string>() { "Text", "ALongerBitOfText" },
                    SelectedIndex = 0,
                    VerticalOptions = LayoutOptions.CenterAndExpand,
                    HorizontalOptions = LayoutOptions.CenterAndExpand,
                },
            };
        }
```
Android - does not resize, instead the size seems fixed based on the initially selected item:
![image](https://user-images.githubusercontent.com/10667683/113273826-ef460e80-92d4-11eb-82fd-077faefe95c6.png)

UWP - works as expected:
![image](https://user-images.githubusercontent.com/10667683/113274076-2fa58c80-92d5-11eb-96eb-b31df13848d6.png)
